internal/cli/kraft/run/runner_kernel.go

```diff
@@ -65,16 +68,22 @@ func (runner *runnerKernel) Prepare(ctx context.Context, opts *RunOptions, machi
 
 		switch fe.Machine {
 		case elf.EM_X86_64, elf.EM_386:
-			machine.Spec.Architecture = "x86_64"
+			machine.Spec.Architecture = arch.ArchitectureX86_64.String()
 		case elf.EM_ARM:
-			machine.Spec.Architecture = "arm"
+			machine.Spec.Architecture = arch.ArchitectureArm.String()
 		case elf.EM_AARCH64:
-			machine.Spec.Architecture = "arm64"
+			machine.Spec.Architecture = arch.ArchitectureArm64.String()
 		default:
 			return fmt.Errorf("unsupported kernel architecture: %v", fe.Machine.String())
 		}
 	} else {
 		machine.Spec.Architecture = opts.Architecture
+
+		if _, found := arch.ArchitecturesByName()[opts.Architecture]; !found {
+			log.G(ctx).WithFields(logrus.Fields{
+				"arch": opts.Architecture,
+			}).Warn("unknown or incompatible")
+		}
```

Comment:
Just checked out this PR to look at everything in context. Can we move this into `run.go` since it applies to all runners? It should be moved just after the `opts.discoverNetworkController` call.